asQ/allatonce/function.py

```diff
@@ -56,6 +56,7 @@ def __init__(self, ensemble, time_partition, function_space):
         """
         self.time_partition_setup(ensemble, time_partition)
 
+        self.t0 = fd.Constant(0)
```

Comment:
Please can you explain again why the initial time is attached to the Function but the time value at each timestep is attached to the Form?
I don't think I understand what the benefit of keeping them separately is - In the old AllAtOnceSystem they were stored together.

Also see below the comment about `t0` and `time` not being independent (another reason to keep them together)

Reply:
Could you please have a look at diag_preconditioner.py. I am not sure if this is the right way to update t_average. 
I think the current setting does not update it.

Reply:
What are you not sure about with the current setting? It looks OK to me.

Reply:
See comment below. You weren't updating the value of `t_average`, you were replacing it with a new `Constant` every time.

Reply:
Resolved by 0b13d508633b3265112aaebc314031abb2966077